### PR TITLE
E2E framework

### PR DESCRIPTION
**Pull request by shivramsrivastava**
This fix is the reinstated the changes in the PR #25 with the new dep dependency and bazel build.

### CODE REVIEW

**Comment by shivramsrivastava** [comment]
@shashidharatd @deepak-vij

**Comment by irfanurrehman** [comment]
@shivramsrivastava whats your strategy for e2e?
I see that you have copied some k/k e2e test framework files here and then vendoring k/k (probably because that portion of code needs dependency from k/k).. ?
In this way if you anyways end up vendoring k/k then why use the k/k e2e framework as a lib and not copy any code?

**Comment by shivramsrivastava** [comment]
/assign @shashidharatd 
@m1093782566 

Can you please review this change.
I have verified it with 'kubetest' and it works fine.
I will be raising the test infra job for e2e test, it will require this change to run the job.

**Comment by shivramsrivastava** [comment]
@nikita15p

File: test/e2e-poseidon-gce.sh [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+
+# setup the env and correct test directory
+cd test/e2e
+
+# update the poseidon deployment yaml with the correct image
+sed -i "s/gcr.io\/poseidon-173606\/poseidon:latest/gcr.io\/$project\/poseidon-amd64:${BUILD_VERSION}/" $POSEIDON_MANIFEST_FILE_PATH
+
+#Run e2e test
+#go test -v ${SCRIPT_ROOT}/test/e2e/*go -args -kubeconfig=/home/ubuntu/.kube/config -firmamentManifestPath=${FIRMAMENT_MANIFEST_FILE_PATH} -poseidonManifestPath=${POSEIDON_MANIFEST_FILE_PATH}
+go test -v . -ginkgo.v -args -firmamentManifestPath=${FIRMAMENT_MANIFEST_FILE_PATH} -poseidonManifestPath=${POSEIDON_MANIFEST_FILE_PATH}
```
Comment (m1093782566) [inline test/e2e-poseidon-gce.sh]: We can remove this line to stub the e2e test for now.

**Pull request merged**